parachain/modules/relayer-fee-tracker/src/lib.rs

```diff
@@ -0,0 +1,203 @@
+// Copyright (C) Polytope Labs Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod withdrawal;
+
+use ismp::{handlers::validate_state_machine, host::IsmpHost, messaging::Proof};
+pub use pallet::*;
+use pallet_ismp::host::Host;
+use sp_runtime::DispatchError;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+    use ismp::host::StateMachine;
+
+    use crate::withdrawal::{Key, WithdrawalInputData, WithdrawalOutputData, WithdrawalProof};
+    use codec::{Decode, Encode};
+    use ismp::router::{DispatchPost, DispatchRequest, IsmpDispatcher};
+    use pallet_ismp::dispatcher::Dispatcher;
+    use sp_core::H256;
+    use sp_runtime::traits::{IdentifyAccount, Verify};
+    use sp_std::{prelude::*, vec};
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    /// The config trait
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_ismp::Config {
+        /// Origin allowed to add or remove parachains in Consensus State
+        type AdminOrigin: EnsureOrigin<<Self as frame_system::Config>::RuntimeOrigin>;
```

Comment:
```suggestion
        /// Origin allowed to add or remove parachains in Consensus State
        type AdminOrigin: EnsureOrigin<<Self as frame_system::Config>::RuntimeOrigin>;
```

We don't need the admin origin, relayers should be able to permissionlessly accumulate their fees